specs/data_structures.md

```diff
@@ -228,100 +229,107 @@ Merkle trees are used to authenticate various pieces of data across the LazyLedg
 
-### Verifying Annotated Merkle Proofs
+### Binary Merkle Tree Proofs
 
-In addition to the root, leaf, index, and sibling values of a Merkle proof for a plain [binary Merkle tree](#binary-merkle-tree), Merkle proofs for annotated Mekle trees have the sibling field values. Proofs are verified by using the appropriate methods to compute field values.
+| name       | type                          | description                   |
+| ---------- | ----------------------------- | ----------------------------- |
+| `root`     | [HashDigest](#hashdigest)     | Merkle root.                  |
+| `key`      | `byte[32]`                    | Key (i.e. index) of the leaf. |
+| `siblings` | [HashDigest](#hashdigest)`[]` | Sibling hash values.          |
+| `leaf`     | `byte[]`                      | Leaf value.                   |
 
 ## Namespace Merkle Tree
 
-[Messages](#message) in LazyLedger are associated with a provided _namespace ID_, which identifies the application (or applications) that will read these messages when parsing blocks. The Namespace Merkle Tree (NMT) is a variation of the [Merkle Interval Tree](https://eprint.iacr.org/2018/642).
+[Shares](#share) in LazyLedger are associated with a provided _namespace ID_. The Namespace Merkle Tree (NMT) is a variation of the [Merkle Interval Tree](https://eprint.iacr.org/2018/642), which is itself an extension of the [Merkle Sum Tree](https://bitcointalk.org/index.php?topic=845978.0). It allows for compact proofs around the inclusion of exclusion of shares with particular namespace IDs.
 
-The NMT is an annotated Merkle tree with two additional fields and methods that indicate the range of namespace IDs in each node's subtree.
-
-For leaf node of message `m`:
+The base case (an empty tree) is defined as:
 ```C++
-n_min = m_1_l(m) = m.namespaceID
-n_max = m_2_l(m) = m.namespaceID
-v = h(serialize(m))
+n_min = 0x0000000000000000000000000000000000000000000000000000000000000000
+n_max = 0x0000000000000000000000000000000000000000000000000000000000000000
+v = 0x0000000000000000000000000000000000000000000000000000000000000000
 ```
 
-The `namespaceID` message field here is the namespace ID of the message, which is a [`NAMESPACE_ID_BYTES`](consensus.md#system-parameters)-long byte array.
+For leaf node of data `d`:
+```C++
+n_min = d.namespaceID
+n_max = d.namespaceID
+v = h(0x00, serialize(m))
+```
 
-Before being hashed, the [messages](#message) are [serialized](#serialization).
+The `namespaceID` message field here is the namespace ID of the leaf, which is a [`NAMESPACE_ID_BYTES`](consensus.md#system-parameters)-long byte array.
 
 For internal node with children `l` and `r`:
 ```C++
-n_min = m_1_i(height, l, r) = min(l.n_min, r.n_min)
-n_max = m_2_i(height, l, r) = max(l.n_max, r.n_max)
-v = h(l, r) = h(l.n_min, l.n_max, l.v, r.n_min, r.n_max, r.v)
+n_min = min(l.n_min, r.n_min)
+n_max = max(l.n_max, r.n_max)
+v = h(l, r) = h(0x01, l.n_min, l.n_max, l.v, r.n_min, r.n_max, r.v)
```

Comment:
Isn't the `n_min||n_max||` prefix missing here?

Reply:
It's unclear the `n_min` and `n_max` become part of the node here.

Reply:
No, these two are isomorphisms:
- tuple of byte arrays `<n_min, n_max, v>`
- single byte array `n_min || n_max || v`

Reply:
> No, these two are isomorphisms:

sure thing, and I understand what you mean here. Still this will be confusing because that part:
> I'm using the first representation here (i.e. a node has 3 fields: node.n_min, node.n_max and node.v).

is kinda missing in the spec.

Reply:
If you mean to say a node has 3 fields, then it should be mentioned somewhere in the spec. and regarding 
https://github.com/lazyledger/lazyledger-specs/pull/47#discussion_r449089619
those should then be `v.n_min` and `v.n_max` respectively.

Reply:
Should be fixed as of 16e468d.